### PR TITLE
Rename Photon demos and split GitHub workflow

### PR DESCRIPTION
**Pull request by fniephaus**
This renames the Photon demos for consistency. Splitting the workflow makes sure the demos are tested independently from each other.

### CODE REVIEW

File: .github/workflows/graalwasm-photon-micronaut.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      - '.github/workflows/graalwasm-photon-micronaut.yml'
+  pull_request:
+    paths:
+      - 'graalwasm/graalwasm-photon-micronaut/**'
+      - '.github/workflows/graalwasm-photon-micronaut.yml'
+  workflow_dispatch:
+permissions:
+  contents: read
+jobs:
+  photon-micronaut:
```
Comment (jirkamarsik) [inline .github/workflows/graalwasm-photon-micronaut.yml:15]: Minor: We can rename this job to `run` for consistency with the other two workflow files.

**Review by jirkamarsik — APPROVED** [review]
Thanks for splitting up the GitHub workflow!

File: .github/workflows/graalwasm-photon-micronaut.yml (hunk 1 above)
Comment (fniephaus) [reply .github/workflows/graalwasm-photon-micronaut.yml:15]: Well spotted, thanks! Will fix in a follow up PR.

Comment (fniephaus) [reply .github/workflows/graalwasm-photon-micronaut.yml:15]: See https://github.com/graalvm/graalvm-demos/pull/286/commits/3e141cf374433ba78a7fa1c32d0c55327cd3e051

**Pull request merged**